dough-items/src/main/java/io/github/bakedlibs/dough/items/ItemStackEditor.java

```diff
@@ -0,0 +1,119 @@
+package io.github.bakedlibs.dough.items;
+
+import org.bukkit.Color;
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.LeatherArmorMeta;
+import org.bukkit.inventory.meta.PotionMeta;
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Fluent class to apply edits/transformations to an {@link ItemStack} and it's {@link ItemMeta} instance
+ * <p>
+ * All methods in this class which are not getters mutate this instance.<br>
+ * The {@link ItemStack} instance which this class holds on to is never mutated.
+ *
+ * @see #create()
+ * @see #applyTo(ItemStack)
+ */
+@ParametersAreNonnullByDefault
+public class ItemStackEditor {
+
+    private final ItemStack itemStack;
+    private Consumer<ItemMeta> metaTransform = null;
+    private Consumer<ItemStack> stackTransform = null;
+
+    public ItemStackEditor(ItemStack item) {
+        this.itemStack = item.clone();
+    }
+
+    public ItemStackEditor(Material type) {
+        this.itemStack = new ItemStack(type);
+    }
+
+    public ItemStackEditor addFlags(ItemFlag... flags) {
+        return appendMetaConsumer(ItemStackUtil.appendItemFlags(flags));
+    }
+
+    public ItemStackEditor setCustomModel(int data) {
+        return appendMetaConsumer(ItemStackUtil.editCustomModelData(data));
+    }
+
+    public ItemStackEditor setCustomModel(@Nullable Integer data) {
+        return appendMetaConsumer(ItemStackUtil.editCustomModelData(data));
+    }
+
+    public ItemStackEditor setAmount(int amount) {
+        return appendStackConsumer(stack -> stack.setAmount(amount));
+    }
+
+    public ItemStackEditor setColor(Color color) {
```

Comment:
i'm really not a _huge_ fan of setColor but I don't have a better name so feel free to ignore.